recipe/meta.yaml

```diff
@@ -11,45 +21,226 @@ source:
+        - if exist %LIBRARY_LIB%\cmake\boost_python-{{ version }} exit 1        # [win]
+        - if exist %LIBRARY_LIB%\cmake\boost_numpy-{{ version }} exit 1         # [win]
+
+        # regular libraries
+        {% for each_boost_lib in boost_libs %}
+        # presence of shared libs
+        - test -f $PREFIX/lib/libboost_{{ each_boost_lib }}.dylib               # [osx]
+        - test -f $PREFIX/lib/libboost_{{ each_boost_lib }}.so                  # [linux]
+        - if not exist %LIBRARY_BIN%\boost_{{ each_boost_lib }}.dll exit 1      # [win]
+        - if not exist %LIBRARY_LIB%\boost_{{ each_boost_lib }}.lib exit 1      # [win]
+
+        # absence of static libs
+        - test ! -f $PREFIX/lib/libboost_{{ each_boost_lib }}.a                 # [unix]
+        - if exist %LIBRARY_LIB%\libboost_{{ each_boost_lib }}.lib exit 1       # [win]
+        {% endfor %}
+
+        # static-only libraries
+        {% for each_boost_lib in boost_libs_static_only %}
+        # presence of static libs
+        - test -f $PREFIX/lib/libboost_{{ each_boost_lib }}.a                   # [unix]
+        - if not exist %LIBRARY_LIB%\libboost_{{ each_boost_lib }}.lib exit 1   # [win]
+
+        # absence of shared libs
+        - test ! -f $PREFIX/lib/libboost_{{ each_boost_lib }}.dylib             # [osx]
+        - test ! -f $PREFIX/lib/libboost_{{ each_boost_lib }}.so                # [linux]
+        - if exist %LIBRARY_BIN%\boost_{{ each_boost_lib }}.dll exit 1          # [win]
+        - if exist %LIBRARY_LIB%\boost_{{ each_boost_lib }}.lib exit 1          # [win]
+        {% endfor %}
+
+        # test compilation
+        - ./test_lib.sh   # [unix]
+        - ./test_lib.bat  # [win]
+
+  # compatibility output for old naming convention
+  - name: boost-cpp
+    requirements:
+      host:
+        # match host deps to avoid hash collisions between wrappers
+        - icu               # [unix]
+        - xz                # [unix]
+        - bzip2
+        - zlib
+        - zstd
+        - {{ pin_subpackage("libboost", exact=True) }}
+      run:
+        - {{ pin_subpackage("libboost", exact=True) }}
+    test:
+      commands:
+        - echo "tested in other outputs"
+
+  - name: libboost-python
+    script: build-py.sh   # [unix]
+    script: build-py.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage('libboost-python', max_pin='x.x') }}
+    requirements:
+      build:
+        - python                                 # [build_platform != target_platform]
+        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+        - numpy                                  # [build_platform != target_platform]
+        - {{ compiler('cxx') }}
+      host:
+        - python
+        - numpy
+        - {{ pin_subpackage("libboost-headers", exact=True) }}
+      run:
+        - python
+        - {{ pin_compatible('numpy') }}
+        - {{ pin_subpackage("libboost-headers", exact=True) }}
```

Comment:
Rephrasing https://github.com/conda-forge/boost-cpp-feedstock/issues/137#issuecomment-1539573207

```suggestion
```

Reply:
I'm not against loosening the pins on the headers in principle, but I don't know boost well enough to know that they'll always be compatible. For context there was a [response](https://github.com/conda-forge/boost-cpp-feedstock/issues/137#issuecomment-1540009887) on that in the other thread:
> @beckermr: I'd prefer exact pins for now with the idea that we can relax them later if it is working.

-------

> @sdebionne: The need for `devel` packages, not specific to Boost, might be out of scope of this MR though.

Yeah, I think this PR is already very large and so far has the scope as defined in https://github.com/conda-forge/boost-cpp-feedstock/issues/137: merge feedstocks, rename packages, add run-exports.

I'll do my best to ensure it does these things to everyone's satisfaction, but I don't want to overload this with refactoring how `libboost-python` works, adding `-devel` packages (another discussion point for consistency / bikeshedding), etc. That should be left to follow-ups.

Reply:
> I'm not against loosening the pins on the headers in principle

Actually, I suggested to remove the requirement alltogether (and only have it for the `-devel` packages).

> I don't know boost well enough to know that they'll always be compatible

The headers _must_ match the version of the library.

> adding -devel packages (another discussion point for consistency / bikeshedding)

Ok, let's discuss this in a new issue.

Reply:
> Yeah, I think this PR is already very large and so far has the scope as defined in https://github.com/conda-forge/boost-cpp-feedstock/issues/137: merge feedstocks, rename packages, add run-exports and an output for only the headers.

I'm against this line of thinking. As I said earlier, there's no need to keep technical debt forever. I agree with @sdebionne's suggestion here.

Reply:
> As I said earlier, there's no need to keep technical debt forever.

There's also no need to require one PR to clean up all extant technical debt. In fact, that's not a reasonable demand _at all_. I'm keeping the commit history sane, but the more this (or any other large) PR does at once, the harder it will be to reason about if anything changes or breaks.

Standard development practices require PRs to be focused on one thing for a reason.

I've taken on your request of where to place the headers for the python libs, as I think it's broadly in scope of this PR. But changing what `libboost-python` pins or brings along is another question. You cannot compile against it without the headers. So if it's then about having a lighter run-dependence, that's a separate improvement.

Because deferring the discussion of how to pin the headers vs. `libboost-python` is in no way "keeping technical debt forever". It's just one PR away after this one gets merged and doesn't blow up anything. Plus I agree with @beckermr that it's much easier to loosen requirements afterwards.

Reply:
> Standard development practices require PRs to be focused on one thing for a reason.

For a bit more context: I recognize of course that in contrast to any upstream repo merging a PR, merging this one will publish artefacts. However, the PR is aimed as much as possible at not breaking anything, just changing behind the scenes through which output a given file gets installed.

In other words, recipes or other consumers of `boost-cpp` should get the exact same files as before[^1]. For `boost`, we've now removed the non-python libraries, but if that's a concern I'm happy to revert this (765cc2cf9d24373bb990ebc82d04b99ebee0d0d5).

[^1]: plus the headers for boost python, after implementing your request for exactly that

This PR aims to achieve a milestone, not the end goal. And shuffling files between outputs with very low risk of breakage (especially as 1.82 is only 2 weeks old and not migrated yet) is IMO a very good way point to achieve before embarking on further improvements. That's why I'm so reluctant to layer more changes on top here that aren't absolutely necessary for the [goals](https://github.com/conda-forge/boost-cpp-feedstock/issues/137) that were agreed upon. Not because I'm lazy, but because I think it's detrimental to maintenance later as well as actually pulling off this large tech debt clean-up.

But if this becomes the stumbling block, then perhaps we publish these builds to a `boost_dev` label, so that we can test them in selected feedstocks and iterate as necessary (but with intermediate way points & easily testable artefacts).

In any case, it would be beneficial to write up a plan of action of what else you think should be done (whether compatibly or incompatibly with the current boost setup; as I keep saying this PR is aiming at the former), rather than saying "this doesn't address all of our technical debt" in this PR - I hope we can agree it's a step in the right direction at least, and that we can build on that.

> I've taken on your request of where to place the headers for the python libs, as I think it's broadly in scope of this PR. 

This is passing now.